specs/phase0/fork-choice.md

```diff
@@ -68,10 +73,34 @@ Any of the above handlers that trigger an unhandled exception (e.g. a failed ass
 ```python
 @dataclass(eq=True, frozen=True)
 class LatestMessage(object):
-    epoch: Epoch
+    slot: Slot
     root: Root
 ```
 
+#### `BlockSlotNode`
+
+```python
+class BlockSlotNode(Container):
+    block_root: Root
+    slot: Slot
+    parent_node: Root
```

Comment:
:+1: about 1.

For 2, I don't think we should use the `BlockSlotKey`/`BlockRootAndSlot` anywhere that it's not needed. The only reason for defining this class is to feed the block root & slot into `hash_tree_root`. `hash_tree_root` is used here so that a new hash function is not introduced in the spec repo, but clients could very well use any other unique identifiers.